Lab4/src/ca/mcgill/ecse211/lab4/Lab4.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * {@value #leftMotor}  value from the left motor
+ * {@value #rightMotor} value from the right motor
+ * {@value #WHEEL_RAD}  constant, value of the wheel's radius
+ * {@value #TRACK}      constant, value of track
+ * {@value #SPEED}      constant, moving speed of robot
+ * {@value #usPort}     value from usPort
+ * 
+ * @author Maxime Bourassa
+ * @author Violet Wei
+ *
+ */
+public class Lab4 {
+
+  // Motor Objects, and Robot related constant parameters
+  //These all have getters to be accessed from other classes
+  public static final EV3LargeRegulatedMotor leftMotor =
+      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("D"));
+  public static final EV3LargeRegulatedMotor rightMotor =
+      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("A"));
+  public static NavigationLab4 Nav = new NavigationLab4();
+  public static final TextLCD LCD = LocalEV3.get().getTextLCD();
+  public static Odometer odometer = new Odometer();
+  public static final double WHEEL_RAD = 2.1;
+  public static final double TRACK = 11.1;
+  public static final int SPEED = 75;
+
+  /**
+   * Localization type that can be used
+   * <li>{@link #FALLING_EDGE}</li>
+   * <li>{@link #RISING_EDGE}</li>     
+   */
+  public enum LocalizationType { FALLING_EDGE, RISING_EDGE };
+
+  private static final Port usPort = LocalEV3.get().getPort("S1");
+  @SuppressWarnings("resource") // Because we don't bother to close this resource
+  public static SensorModes usSensor = new EV3UltrasonicSensor(usPort); // usSensor is the instance
+  public static SampleProvider usDistance = usSensor.getMode("Distance"); // usDistance provides samples provider from this instance
+  public static float[] usData = new float[usDistance.sampleSize()]; // usData is the buffer in which data is
+  // returned in
+
+  /**
+   * Main entry of the program. Starts the run and lets the user select between
+   * different options
+   * 
+   * @param args
+   * @throws OdometerExceptions
+   */
+  public static void main(String[] args) throws OdometerExceptions {  
+
+    //Parameter used in the method
+    int buttonChoice;
+    // implementation
+    DisplayLab4 odometryDisplay = new DisplayLab4(); // No need to change
+
+    do {
+      // clear the display
+      LCD.clear();
+
+      // ask the user what program he wants to run
+      LCD.drawString("   Press left        ", 0, 0);
+      LCD.drawString(" for Falling edge    ", 0, 1);
+      LCD.drawString("   Press right       ", 0, 2);
+      LCD.drawString(" for Rising edge     ", 0, 3);
+      LCD.drawString(" Left to test dist   ", 0, 4);
+      LCD.drawString("    Press down       ", 0, 5);
+      LCD.drawString(" to test rotation    ", 0, 6);
+
+      buttonChoice = Button.waitForAnyPress(); //wait for the user
+    } 
+    while (buttonChoice != Button.ID_ENTER && buttonChoice != Button.ID_DOWN && buttonChoice != Button.ID_LEFT && buttonChoice != Button.ID_RIGHT); //Exit when a valid option has been selected

```

Comment:
This line is waaaaaaaaaaaaaaaaay too long! Know that in Java, line wrapping is acceptable (and even required in this case). Line wrapping means writing one logical line of source code (ending with `;`) on multiple text editor lines, to make the code easier to read. For instance, this code is valid:

```java
System.out
    .println(
        "Hello!"
    );
```
In addition, you could have hovered over `ID_ENTER`, `ID_DOWN`, etc and realized that you can rewrite your code as:

```java
} while (buttonChoice < 0);
```